Dockerfile

```diff
@@ -0,0 +1,11 @@
+FROM alpine:3.13.3
+
+RUN apk add --no-cache git && \
+    mkdir /workdir
+
+COPY git-chglog /usr/local/bin/git-chglog
```

Comment:
I want to note that this works due to a "side effect" that we (and goreleaser) run `make build` on a 64-bit version of Linux. As a result, we can use the `make build` command outside of the container and then `COPY` it into the `alpine` image, 💥  it works. 

A use case to consider: If a user attempts to build the image locally on a Mac or Windows host, the resulting docker image will throw a runtime error due to an incompatible architecture.

If we wanted to be pedantic, we could make a target that specifically calls out that we are creating the binary for the purpose of using within an `apline` image.

```make
alpine: clean
    env GOOS=linux GARCH=amd64 CGO_ENABLED=0 go build -ldflags=$(LDFLAGS) -o git-chglog ./cmd/git-chglog
	
docker: alpine
    docker build --tag quay.io/git-chglog/git-chglog:local .
```

Just something to consider.

Reply:
This, actually, was the main motivation for me on Makefile refactoring, because I was trying to test this locally and noticed `make build` doesn't statically build the binary which has been [modified](https://github.com/git-chglog/git-chglog/blob/master/Makefile#L42) since.

That said in the snippet above `alpine` is not necessarily needed, but `docker` is very nice to have and very convenient. :+1:

Reply:
Are you okay with ?
`
docker: build
    docker build --tag git-chglog .
`

It will be easier to use locally and i could leverage that on the CI

Reply:
Right on. If we wanted to ensure that the `GOHOST` values are correct for binary used with the the `docker` target, what process do you suggest?

The `make alpine` was just an attempt to encapsulate the generation of the binary used in the docker image. Is there a set of parameters we could pass to `make docker` that would not require the `alpine` target? The modifications that first come to mind are adding a build step to the `make docker` target, but that is just the "hammer sees nail" solution 🤣

Reply:
This looks better IMO (somewhere in `Build` section of `Makefile`):

```Makefile
.PHONY: docker
docker: DOCKER_TAG ?= dev
docker: build   ## Build Docker image
	@ $(MAKE) --no-print-directory log-$@
	docker build --pull --tag quay.io/git-chglog/git-chglog:$(DOCKER_TAG) .
```

and then you can update `publish.yml` accordingly: `DOCKER_TAG=edge make docker`